projects/omang-service-mediator/src/modules/mpi/services/mpi.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { AxiosRequestConfig } from 'axios';
+import { Agent } from 'https';
+import { config } from 'src/config';
+import { Bundle } from 'fhir/r4';
+
+@Injectable()
+export class MasterPatientIndex {
+  private readonly logger = new Logger(MasterPatientIndex.name);
+  private readonly clientRegistryUrl: string;
+  private readonly authHeader: string;
+  private readonly devMode: boolean;
+
+  constructor(private readonly httpService: HttpService) {
+    this.clientRegistryUrl = config.get('ClientRegistry:ApiUrl');
+    const authString = ''; //`${client}:${password}`;
+    this.authHeader = `Basic ${Buffer.from(authString).toString('base64')}`;
+    this.devMode = config.get('ClientRegistry:devMode') === 'true';
+  }
+
+  private getHttpOptions(): AxiosRequestConfig {
+    const options: AxiosRequestConfig = {
+      headers: {
+        Authorization: this.authHeader,
+        'Content-Type': 'application/fhir+json',
+      },
+    };
+
+    if (this.devMode) {
+      options.httpsAgent = new Agent({
+        rejectUnauthorized: false,
+      });
+    }
+
+    return options;
+  }
+
+  async getSearchBundle(query): Promise<any> {
+    try {
+      const searchResponse = await this.httpService.axiosRef.get<Bundle>(
+        `${this.clientRegistryUrl}/Patient`,
+        {
+          params: { identifier: query },
+          headers: {
+            'Content-Type': 'application/fhir+json',
+            'x-openhim-clientid': 'OmangSvc'
+          },
+        },
+      );
+
+      return searchResponse.data;
+    } catch (error) {
+      this.logger.error(
+        'Could not get CR bundle for patient with ID ' +
+          query +
+          '\n' +
+          error,
+      );
+    }
+  }
+
+  async createPatient(patient: any): Promise<any> {
+    try {
+      // Fix for date default formatting issue in FHIR SDK
+      if (patient.birthDate) {
+        patient.birthDate = new Date(patient.birthDate)
+          .toISOString()
+          .split('T')[0];
+      }
+      delete patient.id;
+
+      const response = await this.httpService.axiosRef.post(
+        `${this.clientRegistryUrl}/Patient`,
+        patient,
+        {
+          headers: {
+            'Content-Type': 'application/fhir+json',
+            'x-openhim-clientid': 'OmangSvc'
```

Comment:
Assuming this clientID is not something that might change? or should we consider making this a variable that can be supplied in case it changes

Reply:
This is something that I added cause OpenCR requires it. I need to check with Piotr what the intended use. It may be that this needs to be one of the source systems : Omang Registry, BDRS Registry, OpenMRS, ...

Reply:
aah ok cool, Yeah im wondering if this shouldnt be the source system making the request via OpenHIM, so this will change depending on which systems sends the request. But maybe confirmation is needed forst before changing this :+1: